src/utils.rs

```diff
@@ -1,9 +1,151 @@
 //! Utility date and time equations for Temporal
 
-use crate::MS_PER_DAY;
+use num_traits::{AsPrimitive, FromPrimitive};
+
+use crate::{TemporalError, TemporalResult, MS_PER_DAY};
 
 // NOTE: Review the below for optimizations and add ALOT of tests.
 
+#[derive(Debug, Default, Clone, Copy, PartialEq, PartialOrd)]
+pub struct FiniteF64(pub(crate) f64);
+
+impl FiniteF64 {
+    #[inline]
+    #[must_use]
+    pub unsafe fn new_unchecked(value: f64) -> Self {
+        Self(value)
+    }
```

Comment:
Does this need to be unsafe? We're not applying any memory optimizations that depend on f64 being finite, right?

Reply:
Not necessarily for memory optimizations, but the idea of it marking it as unsafe was primarily because it's an unchecked construction. For instance, someone might do something like:

```rust
let totally_finite = FiniteF64::new_unchecked(f64::NAN);
```

It totally undercuts the invariance of the type, so to make the unsafe is meant more to mark that unchecked API public. That being said, I'd be fine with just getting rid of the method entirely. As I'm not entirely sold on having the method in general (I had started with it there, and as I worked on this, the less I thought it was needed).